### PR TITLE
zfs modules: adjust docs

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY

Adjustments to the module documentation

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

zfs
zfs_delegate_admin
zfs_facts

### CODE REVIEW

File: plugins/modules/zfs_facts.py [hunk 1]
```diff
@@ -21,140 +20,137 @@
...
+  type: str
+  sample: rpool/var/spool
+parsable:
+  description: if parsable output should be provided in machine friendly format.
+  returned: if 'parsable' is set to True
+  type: bool
+  sample: true
+recurse:
+  description: if we should recurse over ZFS dataset
+  returned: if 'recurse' is set to True
```
Comment (russoz) [inline plugins/modules/zfs_facts.py:87]:
```suggestion
  returned: if O(recurse=True)
```

File: plugins/modules/zfs_facts.py [hunk 2]
```diff
@@ -21,140 +20,137 @@
...
+
+RETURN = r"""
+name:
+  description: ZFS dataset name
+  returned: always
+  type: str
+  sample: rpool/var/spool
+parsable:
+  description: if parsable output should be provided in machine friendly format.
+  returned: if 'parsable' is set to True
```
Comment (russoz) [inline plugins/modules/zfs_facts.py:82]:
```suggestion
  returned: if O(parsable=True)
```

File: plugins/modules/zfs.py [hunk 3]
```diff
@@ -53,47 +49,47 @@
...
-- name: Destroy a filesystem
-  community.general.zfs:
-    name: rpool/myfs
-    state: absent
-'''
+  - Johan Wiren (@johanwiren)
+"""
+
+EXAMPLES = r"""
+  - name: Create a new file system called myfs in pool rpool with the setuid property turned off
```
Comment (felixfontein) [inline plugins/modules/zfs.py:56]: Same here.

**Comment by russoz** [comment]
@felixfontein apparently the dev version of the sanity tests are flunking, unrelated to this PR

File: plugins/modules/zfs.py [hunk 4]
```diff
@@ -9,23 +9,20 @@
 from __future__ import absolute_import, division, print_function
 __metaclass__ = type
 
-DOCUMENTATION = '''
----
+DOCUMENTATION = r"""
 module: zfs
 short_description: Manage zfs
```
Comment (felixfontein) [inline plugins/modules/zfs.py:14]: While you're at it, can you fix the casing here?
```suggestion
short_description: Manage ZFS
```

**Review by felixfontein — COMMENTED** [review]
Besides that, LGTM :)

**Review by felixfontein — APPROVED** [review]

**Pull request merged**